### DISCUSSION TITLE
👋 I'm Back!😅

### DISCUSSION THREAD

**Discussion by Jyers** [Jyers/Marlin#1513] [General Discussion]
Hey!

So yeah, long time no see 😅
Long story short, college is quite a time eater! On top of which I've also been putting off finishing the transition to the new bugfix code. The funny part is I did more than half the transition a month ago, and then exams happened and I put it off another month...

But now it's done!
So big news is that the main branch has changed, as you all probably noticed. Thanks to my PR some time ago, this fork is now 100% based off the official marlin bugfix code. With the only differences currently being my most recent feature additions and some fixes. This means not only will it be significantly easier to keep up to date with the upstream code and get fixes (even for the display code!) but it also means I can push all my new feature additions upstream as soon as they are tested! Hopefully that's at least half as exciting to you as it is to me, because I'm stoked!

Hopefully now that I have that out of the way I can put a little more time into updating the firmware. I know it's probably been feeling a bit stale to you all after having to go months without any updates or fixes. But rest assured, I have new features in the works!

Just a note, there may be some major bugs in the current code, as I have heard of some strange ones upstream. It may take a week or so to work them out, but hopefully there are plenty of you willing to plunge in right away and help us get everything ironed out. I'll make a release soon so anyone who wants to can lend a hand if they so choose.

I just want to thank everyone so much for sticking with the project even when it slowed down. This community is something very dear to my heart and the fact that it's still kicking makes immensely happy. You all are essential in making this firmware what it is, never forget that.

Now to go through 4 pages of unanswered discussion posts...

Best,
Jyers

**Note:** Just realized I completely forgot to mention an important change due to the new branch. Configs will need some tweaking. I'd recommend starting back from one of the templates if possible, or if you have lots of changes, do a diff check and just carry your changes into the new template. This UI now has a new define labeled JyersUI instead of creality dwin lcd to allow it to exist upstream among the others as well as there being some conditional changes throughout the config.

**Comment by PapsDizzy** [comment]
Welcome back mate

**Reply by gothcha** [reply]
Pleased that you are over the exams and are still up for the fight! Look forward to seeing what you are planning 👍

**Comment by SikVac** [comment]
Much success in creating new versions of this excellent program, can't wait for the promised new features.

**Comment by patofoto** [comment]
Welcome back!. Happy to hear college is going well and that your priorities are set straight.

**Comment by Cal-LF** [comment]
Happy you are back!

**Comment by Shinevision** [comment]
Awesome!!
Welcome back!

**Comment by gothcha** [comment]
From all the relieved messages, I think maybe you should be called Lazarus not Jacob 🤣

**Comment by jetel73** [comment]
Nice to see you're back. Hope your exams are a success.

**Reply by Jyers** [reply]
I hope so too 😅

**Comment by Japander** [comment]
Thank you for improving this firmware more and more, its a delight :)

**Comment by RayPS** [comment]
Welcome back!

**Comment by HiggsWRX** [comment]
I can't imagine owning an E3V2 without your firmware, you're doing an amazing work and providing us with such a good firmware, for free!

Can't thank you enough!

**Comment by vick-vinegar** [comment]
Do I need to reflash the LCD with this update? Or will the former Jyers LCD UI work fine?

**Reply by Jyers** [reply]
The former DWIN_SET will work with the new code just fine :)

**Reply by saskford** [reply]
I am fairly new to your firmware and have not heard about a DWIN_SET to flash the LCD. Where do I find it and what changes to the display should I expect to see? Great work on the firmware, beats all the others I have tried by a long shot.

**Reply by vick-vinegar** [reply]
Thank you for the help. The firmware is excellent.

**Comment by Tuncay-Ayhan** [comment]
```
Welcome back!
```

**Comment by samatild** [comment]
I'm writing just to thank you for all the effort on this Project. My E3v2 wouldn't be the same without Jyers...
College is a time eater, but it's worthy.  Wish you all the success and we can wait for upcoming releases 👍  
...going to install v2.0.0 on my boards now!

**Comment by minouska** [comment]
No problem loading the new firmware and all functions appear ok, but constant re-boot when trying to print a file!
Any idea about this major problem?
Thanks.

**Reply by Jyers** [reply]
Check out #1522

Should be fixed in the latest source

**Reply by andreasse** [reply]
I downloaded the precompiled code 14 horus ago and I also have this problem. I'm a bit new to github, what does the above comment mean? Does it mean that I need to compile it myself for it to work, and better use an older version if I want to use a precompiled file?

**Comment by techtosterone** [comment]
Good stuff, Jake. Welcome back.

**Comment by NoxImus421** [comment]
Hey Jyers, could you do a microswiss direct drive home offset for the ender 3 v2? I can donate for your time. 10mm offsets x and y would be perfect:)

**Reply by Jyers** [reply]
Unfortunately doing individual compiles is a bit out of my time schedule. You can either ask around and see if a kind community member will help you, or become a patreon. As a patreon you'll get access to my auto-compile service which can compile custom firmware for you based on full marlin configs, or a simplified config I developed.

Hopefully this helps :)

**Reply by Jyers** [reply]
Just read this again, you can also simply do this from the firmware. In Control>Motion>Home Offsets you can set your x y home offsets

**Reply by NoxImus421** [reply]
> Just read this again, you can also simply do this from the firmware. In Control>Motion>Home Offsets you can set your x y home offsets

I've tried that a few times, I've saved it, and then cycled the printer. Unfortunately it doesn't seem to save or recognize it I guess. I turn the machine back on, and run a home operation, or print and it doesn't work. Though I can see it there in the menu.

**Comment by Jer406** [comment]
Dont know if i keep missing it but has the ability to turn off the beeper been removed? 
Awesome work as always. Ill never run this printer without your firmware :D

**Reply by Jer406** [reply]
Thanks for the info. ill look into it

**Comment by MicroSur** [comment]
Are u sure that come back? )
Any news will be welcome. released FW so buggy.

**Comment by PapsDizzy** [comment]
clearly personal things and family things take precedence. but this post that you're back and then gone just as quickly is not nice. A short info about the fact that there is no time would be good, so you hang in the air and know how to go on. I hope jyers is fine and there are no bad reasons that he is gone again.

**Comment by mrrstrat** [comment]
Maybe he can enlist the help from a few of us - a couple of regression testers, a few development team members, a couple people to track bugs and release details.
http://mrc-tech.net

**Comment by mrrstrat** [comment]
@tititopher68-dev  am sure he (@Jyers)  is not offended by people trying to fan the fire of this project and sustain it - probably just bogged down I would imagine if he is still in school.

I remember my undergrad days trying to juggle fun projects with school. After school (and graduate school) the real work began and then there was no time for about 10 years. I was in my early 30s before I was able to write  the Prologic4 language and Dev system as work was a blur of back-to-back projects that never stopped at work. Life of a software developer these days is more of a tool than back 30 years ago - we were the rockstars of business. Now the suits want tech people to sit in a pod away from "the people" and get fed requirements over a wall from mildly technical well meaning analysts - and jam that work in two week blocks and call it "Agile". It always amuses me to see history repeat itself at the cost of evolved sensibility.

Is this project his senior project for school? The issue is that this is now a popular Marlin branch for a lot of us. I have messed with the TH3D and vanilla Marlin code as well and like the Jyers spin - I am using it for all of the machines for the SBA training lab I run and my own company development machines. I suspect others are using it as well because it is pretty slick :-)

Constraints, speed & feed, and new ideas are the stuff spin-off branches are made of - I can help sustain this branch (Jyers spin) as I have with the Cura project but I have limits on time too. But a reasonably regulated project with a part-time army can do it.

Teams I have ever managed were sometimes matrixed and had divided attention - but a part-time team  (a Jyers army if you will) with these roles could help Mr Jyers tremendously:

- release management team members
- documentation team members
- testing & regression testing team members
- core development team members
- bug swatters and support team members (other team members can double as these if needed)
- Meet at some cadence for direction, bug status, release planning

This release also targets the E3 V2 and standard monitor (not the touch screen) from what I can tell - I have not dug into it too deep other than to tool it for my purposes. I am using it successfully with a Ender 3 Max as well (basically is a E3V2). How about the other Ender machines, or the newest Creality machines? How about other boards/monitor combinations?

If this project is tightly coupled to the Ender 3 V2 then its programmed life is as such.

**Reply by marcop300** [reply]
@mrrstrat i like your write up. if you can give me GREAT advise on what code ver to use (that i do not have to compile) . I am ALL EARS on what BIN to use on my Ender 3d V2 4.2.2 with the new DACAI Video screen. i also have the BT Touch install and NOTHING else... its 100% stock and my Printer is 11 days old. i dont know if i need the 3x3,5x.. blah blah blah. I would like to hear what you are running and i will download the SAME ONES.. ha ha.. thanks for your time.

**Reply by mrrstrat** [reply]
I would probably use the latest branch (~/releases on the GitHub link of the code for Jyers) and download the pre-compiled binaries. It is not so hard to compile your own but if you are not comfortable doing so can be somewhat dangerous for your printer :-)

Use the BL touch versions (3x3 or 5x5). These also work on the CR touch as well. On my Ender 3 Max with CR touch I did have to zero in where the home point was (messing with the firmware parameters and measuring where "home" is for that model as it is different than the Ender 3 V2).

**Reply by marcop300** [reply]
so far SPOT on.  Thank you. I am using @tititopher68-dev branch of Marlin the latest one he has. as FW, then I also have the Firmware display of Gotcha2 as a private DIR to install the DACAI video screen. I did use the 5x5 BT file.. BLTouch-5x5-v4.2.2-v2.0.1.bin. . I am on my 1st print today  (that little boat). I also am LEARNING SOOO much about bed leveling....etc I also am using OctoPrint (that is VERY COOL) with a RasPI and a USB Cam.  Thank you for your time mrrstrat. I am sure going to be HOOKED on this printer. I have NO IDEA what I NEED to print. But I will be learning using Cura and TinkerCAD..etc thank you again
